unity/Assets/Scripts/BaseFPSAgentController.cs

```diff
@@ -2597,6 +2598,79 @@ bool includeInvisible
+            }
+
+            var objectIdToConvexHull = new Dictionary<string, List<(float x, float y)>>();
+            foreach (var sop in GameObject.FindObjectsOfType<SimObjPhysics>()) {
+                if (
+                    allowedObjectTypesSet != null 
+                    && !allowedObjectTypesSet.Contains(sop.Type.ToString())
+                ) {
+                    continue;
+                }
+                var vertices = new List<MIConvexHull.DefaultVertex2D>();
+                float maxY = -float.PositiveInfinity;
+
+                foreach (var meshFilter in sop.GetComponentsInChildren<MeshFilter>()) {
+                    foreach (var localVertex in meshFilter.mesh.vertices) {
+                        var globalVertex = meshFilter.transform.TransformPoint(localVertex);
+                        vertices.Add(new MIConvexHull.DefaultVertex2D(globalVertex.x, globalVertex.z));
+                        maxY = Math.Max(maxY, globalVertex.y);
+                    }
+                }
+                var miconvexHull = MIConvexHull.ConvexHull.Create2D(vertices, 1e-10);
+                objectIdToConvexHull[sop.ObjectID] = miconvexHull.Result.Select(p => ((float) p.X, (float) p.Y)).ToList();
+
+                #if UNITY_EDITOR
+                Debug.Log(sop.ObjectID);
+                var pointsOnHullArray = miconvexHull.Result.ToArray();
+                for (int i = 0; i < pointsOnHullArray.Length; i++) {
+                    var p0 = pointsOnHullArray[i];
+                    var p1 = pointsOnHullArray[(i + 1) % pointsOnHullArray.Length];
+                    Debug.DrawLine(
+                        new Vector3((float) p0.X, maxY, (float) p0.Y),
+                        new Vector3((float) p1.X, maxY, (float) p1.Y),
+                        Color.red, 100.0f
+                    );
+                }
+                #endif
+            }
+            actionFinishedEmit(true, objectIdToConvexHull);
+        }
+        
+        public void GetPixelToWorldPoint(float row, float col, float depth) {
+            actionFinishedEmit(
+                true, 
+                m_Camera.ScreenToWorldPoint(
+                    new Vector3(col, m_Camera.pixelHeight - row, m_Camera.nearClipPlane)
+                )
+            );
+        }
+
+        public void CameraRayCast(float x, float y) {            
+            Ray ray = m_Camera.ViewportPointToRay(new Vector3(x, y, 0.0f));
+            RaycastHit hit;
+            Physics.Raycast(
+                ray,
+                out hit,
+                Mathf.Infinity,
+                1 << 0 | 1 << 8 | 1 << 10 | 1 << 11,
+                QueryTriggerInteraction.Ignore
+            );
+            #if UNITY_EDITOR
+            Debug.DrawLine(
+                hit.point,
+                hit.point + new Vector3(0f, 1f, 0f),
+                Color.red, 
+                100.0f
+            );
+            #endif
+            actionFinishedEmit(true, hit.point);
```

Comment:
What is the use case from Python for determining the `(x, y, z)` coordinate that is hit from `(x, y)` on the last image frame? Maybe determine distance to the point that is in view(?)

Reply:
Action removed.